### PR TITLE
ena: 2.5.0 -> 2.7.1 & nixos/amazon-image: default to 5.15 kernel 

### PR DESCRIPTION
**Pull request by euank**
###### Description of changes

This bumps the ENA driver, and removes a change that overrode the default kernel for the amazon-image to 5.10. Since the default is 5.15 for nixos as a whole, and 5.15 now works, we can let the amazon-image use the default.

 This update is supposed to give us support for the 5.17 kernel according to the
 2.7.0 release notes here: https://github.com/amzn/amzn-drivers/releases/tag/ena_linux_2.7.0

 Unfortunately, in practice, it does not build with `linuxPackages_5_17`,
 so I left that marked as broken.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [x] Tested, as applicable:
    Manually tested this by booting an ec2 instance and `nixos-rebuild switch`ing through `linuxPackages_5_15` and `5_10` with this change pulled in. I verified the network kept working, including after a reboot, after each kernel version update.
    My methodology was basically to boot the default nixos ami and do `nix-channel --add "https://github.com/euank/nixpkgs/archive/856c922b33c5c3f7de6b55872887077ece431854.tar.gz" nixos && nix-channel --update`, and swap through each kernel version in `configuration.nix`
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by zowoq — APPROVED** [review]
I don't use this myself, I was just unblocking the 5.15 kernel bump in https://github.com/NixOS/nixpkgs/pull/158919. Changes seem okay, `ena` / `amazonImage` build but that is as far as I can test it.

**Comment by euank** [comment]
Thanks for the review!

I'll add @edolstra since they're marked as the maintainer of the package. Originally, I didn't want to add more on their already full plate, but since there's not another maintainer listed..

I do feel that my testing (as described in the checklist) is probably enough to consider this safe, but I don't mind waiting on a maintainer to sign off either.

**Comment by Artturin** [comment]
It'd be nice if there was a test for this

**Pull request merged**
